lib/Driver/ToolChains.cpp

```diff
@@ -1054,8 +1076,21 @@ toolchains::Darwin::constructInvocation(const LinkJobAction &job,
   // relative to the compiler.
   SmallString<128> RuntimeLibPath;
   getRuntimeLibraryPath(RuntimeLibPath, context.Args, *this);
+
+  // Link the standard library.
   Arguments.push_back("-L");
-  Arguments.push_back(context.Args.MakeArgString(RuntimeLibPath));
+  if (context.Args.hasFlag(options::OPT_static_stdlib,
+                            options::OPT_no_static_stdlib,
+                            false)) {
+    SmallString<128> StaticRuntimeLibPath;
+    getRuntimeStaticLibraryPath(StaticRuntimeLibPath, context.Args, *this);
+    Arguments.push_back(context.Args.MakeArgString(StaticRuntimeLibPath));
+    Arguments.push_back("-lc++");
+    Arguments.push_back("-framework");
+    Arguments.push_back("Foundation");
+    Arguments.push_back("-force_load_swift_libs");
+  }
+  else Arguments.push_back(context.Args.MakeArgString(RuntimeLibPath));
```

Comment:
Coding convention nitpick: please put the `else` on the same line as the close brace, and the body on the following line. I usually put braces around a single-line else-clause if the then-clause also had braces, too.